src/schema/api/Entry.h

```diff
@@ -0,0 +1,42 @@
+#ifndef LOODB_ENTRY_H
+#define LOODB_ENTRY_H
+
+#include "EntryType.h"
+#include <memory>
+
+// Private members of Entry
+struct EntryImpl {
+  EntryType type_;
+};
+
+class Entry {
+
+    // pImpl for Entry
+    std::unique_ptr<EntryImpl> impl_;
+
+  public:
+
+    // Constructor
+    Entry( /* params */ );
+
+    // Overload less than operator
+    bool operator<(const Entry&);
+
+    // Overload less than or equals operator
+    bool operator<=(const Entry&);
+
+    // Overload greater than operator
+    bool operator>(const Entry&);
+
+    // Overload greater than or equals operator
+    bool operator>=(const Entry&);
+
+    // Overload equals operator
+    bool operator==(const Entry&);
+
+    // Overload not equals operator
+    bool operator!=(const Entry&);
```

Comment:
With respect to the overloaded operators, maybe focus on the `<` and `!=`. The rest may add room for error, but I'll leave it at your discretion.

Reply:
I think we should have them for convenience, but we may not even need them at all. So we'll see.